ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Genesis/Governor.hs

```diff
@@ -253,7 +270,55 @@ sharedCandidatePrefix curChain candidates =
       -- 'InvalidIntersection' within that ChainSync client, so it's
       -- sound to pre-emptively discard their candidate from this
       -- 'Map' via 'mapMaybe'.
-      Map.mapMaybe splitAfterImmutableTip candidates
+      mapMaybe splitAfterImmutableTip disengagedCandidates
+
+    addCSJCandidateSuffix maybePeer (loeFrag, candidateSuffixes) = case maybePeer of
+      Nothing -> (loeFrag, candidateSuffixes)
+      Just (peer, (frag, _)) ->
+        case AF.splitAfterPoint frag (AF.headPoint loeFrag) of
+          Just (_, suffixFrag) -> (loeFrag, (peer, suffixFrag) : candidateSuffixes)
+          -- If the peer's fragment doesn intersect with the current selection,
+          -- the loeFrag won't intersect with it either.
+          Nothing              -> (loeFrag, candidateSuffixes)
+
+    isObjector = \case
+      Objector{} -> True
+      _          -> False
+
+    isDisengaged = \case
+      Disengaged{} -> True
+      _            -> False
+
+    isDynamo = \case
+      Dynamo{} -> True
+      _        -> False
+
+-- | Given the immutable tip and the dynamo fragment, find the youngest common
+-- intersection between the dynamo fragment, the objector fragment, and all of
+-- the jumper fragments.
+getCSJLoeFrag ::
+  (GetHeader blk) =>
+  Point (Header blk) ->
+  AnchoredFragment (Header blk) ->
+  [(ChainSyncJumpingState m blk)] ->
+  Maybe (AnchoredFragment (Header blk))
+getCSJLoeFrag immutableTip dynamoFrag csjStates =
+    let frags = mapMaybe maybeJumperOrObjectorFragment csjStates
+        splitFrags = filter (\frag -> AF.headSlot frag >= pointSlot immutableTip) frags
+        loeFrag = case splitFrags of
+          [] -> dynamoFrag
+          _  -> minimumBy (compare `on` AF.headSlot) splitFrags
+     in snd <$> AF.splitAfterPoint loeFrag immutableTip
```

Comment:
I don't think your counterexample is valid, because frags contains the last good jump of every jumper and objector, and `loeFrag` is picked among them.

Reply:
Maybe you'd like to propose a better name for `maybeJumperOrObjectorFragment` :)

Reply:
Right, I think I was misled by "the objector fragment" in the Haddocks, while it actually is the fragment up to the last jump the objector accepted :+1:

> Maybe you'd like to propose a better name for `maybeJumperOrObjectorFragment` :)

Maybe sth like `lastUnanimousJump`?